src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -887,6 +920,67 @@ public void testGetImportedKeysDW() throws SQLException {
+            getColumnMetaDataClass.put(TABLE_NAME, String.class);
+            getColumnMetaDataClass.put(COLUMN_NAME, String.class);
+            getColumnMetaDataClass.put(DATA_TYPE, Integer.class);
+            getColumnMetaDataClass.put(TYPE_NAME, String.class);
+            getColumnMetaDataClass.put(COLUMN_SIZE, Integer.class);
+            getColumnMetaDataClass.put(BUFFER_LENGTH, Integer.class); // Not used
+            getColumnMetaDataClass.put(DECIMAL_DIGITS, Integer.class);
+            getColumnMetaDataClass.put(NUM_PREC_RADIX, Integer.class);
+            getColumnMetaDataClass.put(NULLABLE, Integer.class);
+            getColumnMetaDataClass.put(REMARKS, String.class);
+            getColumnMetaDataClass.put(COLUMN_DEF, String.class);
+            getColumnMetaDataClass.put(SQL_DATA_TYPE, Integer.class);
+            getColumnMetaDataClass.put(SQL_DATETIME_SUB, Integer.class);
+            getColumnMetaDataClass.put(CHAR_OCTET_LENGTH, Integer.class);
+            getColumnMetaDataClass.put(ORDINAL_POSITION, Integer.class);
+            getColumnMetaDataClass.put(IS_NULLABLE, String.class);
+            getColumnMetaDataClass.put(SCOPE_CATALOG, String.class);
+            getColumnMetaDataClass.put(SCOPE_SCHEMA, String.class);
+            getColumnMetaDataClass.put(SCOPE_TABLE, String.class);
+            getColumnMetaDataClass.put(SOURCE_DATA_TYPE, Short.class);
+            getColumnMetaDataClass.put(IS_AUTOINCREMENT, String.class);
+            getColumnMetaDataClass.put(IS_GENERATEDCOLUMN, String.class);
+            getColumnMetaDataClass.put(SS_IS_SPARSE, Short.class);
+            getColumnMetaDataClass.put(SS_IS_COLUMN_SET, Short.class);
+            getColumnMetaDataClass.put(SS_UDT_CATALOG_NAME, String.class);
+            getColumnMetaDataClass.put(SS_UDT_SCHEMA_NAME, String.class);
+            getColumnMetaDataClass.put(SS_UDT_ASSEMBLY_TYPE_NAME, String.class);
+            getColumnMetaDataClass.put(SS_XML_SCHEMACOLLECTION_CATALOG_NAME, String.class);
+            getColumnMetaDataClass.put(SS_XML_SCHEMACOLLECTION_SCHEMA_NAME, String.class);
+            getColumnMetaDataClass.put(SS_XML_SCHEMACOLLECTION_NAME, String.class);
+        }
+
+        try (Connection conn = getConnection()) {
+            ResultSetMetaData metadata = conn.getMetaData().getColumns(null, null, tableName, null).getMetaData();
+
+            // Ensure that there is an expected class for every column in the metadata result set
+            assertEquals(metadata.getColumnCount(), getColumnMetaDataClass.size());
+
+            for (int i = 1; i < metadata.getColumnCount(); i++) {
+                String columnLabel = metadata.getColumnLabel(i);
+                String columnClassName = metadata.getColumnClassName(i);
+                Class<?> expectedClass = getColumnMetaDataClass.get(columnLabel);
+
+                // Ensure the metadata column is in the metadata column class map
+                assertNotNull(expectedClass);
```

Comment:
This was mentioned in a previous comment above, but ensure that `expectedClass` is not null, don't assert.

Reply:
```suggestion
                if (null != expectedClass) {
```

Reply:
umm, should this fail? is null a failure scenario? if we did the null check then it would just skip

Reply:
did we accidentally remove the check for isAssignableFrom that was in previous commit?

Reply:
The expected class can be null if the the column label/name in the expected class lookup table does not match the actual column label from the result set (i.e. the lookup by column name will fail). This shouldn't happen but it is possible if there was a future code change or perhaps the returned metadata columns change.

Yes the IsAssignableFrom() is no longer used as I changed the BUFFER_LENGTH metadata class from Object to Integer. It was because of this generic Object class that we needed to use the IsAssignableFrom() check. Is The reason why it was originally specified as a generic Object was that the JDBC spec does not use the BUFFER_LENGTH column and so I reflected that as a generic Object being the expected data type. But in the code I have always ensured that BUFFER_LENGTH was an Integer. Therefore to simplify the logic and remove the IsAssignableFrom() check, I switched the expected class for the BUFFER_LENGTH column from Object to Integer.